gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManagementTaskStreamImpl.java

```diff
@@ -53,14 +53,18 @@ public class DagManagementTaskStreamImpl implements DagManagement, DagTaskStream
 
   @Inject(optional=true)
   protected Optional<DagActionStore> dagActionStore;
+  protected ReminderSettingDagProcLeaseArbiter reminderSettingDagProcLeaseArbiter;
   @Inject
   private static final int MAX_HOUSEKEEPING_THREAD_DELAY = 180;
   private final BlockingQueue<DagActionStore.DagAction> dagActionQueue = new LinkedBlockingQueue<>();
 
+  // TODO: need to pass reference to DagProcLeaseArbiter without creating a circular reference in Guice
   @Inject
-  public DagManagementTaskStreamImpl(Config config, Optional<DagActionStore> dagActionStore) {
+  public DagManagementTaskStreamImpl(Config config, Optional<DagActionStore> dagActionStore,
+      ReminderSettingDagProcLeaseArbiter reminderSettingDagProcLeaseArbiter) {
```

Comment:
looks like you have to update guice module with `ReminderSettingDagProcLeaseArbiter`  in this PR only to make tests pass

Reply:
updated guice module for this and all related classes